### ISSUE TITLE
Plugin Exception handling

### ISSUE THREAD

**Issue by cmac3095** [markdeaton/quickcapture_proplugindatasource_net#1]
In some cases the exceptions are getting "swallowed" (is that the word?) when I think they should just be allowed to bubble up into the calling code. The code also has a tendency to catch its own (thrown) exceptions and I don't quite follow that...

Specifically: ProPluginDatasourceTemplate Open. Open should just fail if it fails.
                                    ProPluginDatasourceTemplate OpenTable. Ditto

Generic System.Exception shouldn't be thrown. OpenTable line 194. This looks like an InvalidOperationException? you pick...

ProPluginTableTemplate constructor.
What does it mean if the _attachmentsDir wasn't correctly set and we call `Open()`? (any exception is caught and swallowed).

ProPluginTableTemplate  AddAttributeColumns there are cases when exceptions are caught (null reference??) and rethrown as Generic "System.Exception", line334 exception is caught and swallowed

**Comment by markdeaton** [comment]
1. "ProPluginDatasourceTemplate Open. Open should just fail if it fails." -- agreed; should have cleaned this up before now.

2. "ProPluginDatasourceTemplate OpenTable. Ditto" -- not sure I agree completely. An exception thrown from here is caught in AddQuickCaptureTableToMap (button) OnClick(). All details and inner exceptions are stripped and the user receives an unhelpful "An exception of type System.Exception was thrown". However, I've removed the try/catch from here as well.

3. "OpenTable line 194. This looks like an InvalidOperationException?" -- agreed and done.

4. "What does it mean if the _attachmentsDir wasn't correctly set and we call Open()?" -- should be addressed by the fix to #1 above,  I think.

5. "ProPluginTableTemplate AddAttributeColumns...line334 exception is caught and swallowed"
* Line 334 is an abuse of try/catch to handle an expected condition and has been replaced with an if/else
* Line  346: issues with malformed input JSON will now error up to the calling routine
